src/apps/replication/lib/replication_failure_detector.cpp

```diff
@@ -117,15 +123,9 @@ void replication_failure_detector::end_ping(::dsn::error_code err, const fd::bea
             {
                 ddebug("update meta server leader to [%s]", ack.this_node.to_string());
                 dsn_group_set_leader(_meta_servers.group_handle(), ack.this_node.c_addr());
```

Comment:
it is now necessary. As right now, we don't really use _meta_servers group address as the beacon target yet, but only use it for maintaining the group-leader meta state. We should make it more transparent and use _meta_servers as the parameter for register_master, in the future. In that case, replication_fd is not necessary as the base fd will have this power already:)